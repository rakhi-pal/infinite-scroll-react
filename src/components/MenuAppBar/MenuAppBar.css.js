import { makeStyles } from "@material-ui/core";
const useStyles = makeStyles((theme) => ({
    menu: {
      flexGrow: 1,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      flexGrow: 1,
    },
}));

export {useStyles};