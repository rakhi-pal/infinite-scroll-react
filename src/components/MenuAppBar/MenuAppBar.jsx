import React from 'react';
import { useStyles } from './MenuAppBar.css';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import useLoginToken from '../../customHooks/useLoginToken';


export default function MenuAppBar({history}) {
  const classes = useStyles();
  const {removeToken} = useLoginToken();

  const handleOnClick = event => {
    removeToken();
    history.push('/');
    window.location.reload();
  }

  return (
    <div className={classes.menu}>
      <AppBar position="static"  data-testid="appbar">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            Contacts
          </Typography>
          <Button onClick={e => handleOnClick(e)} color="inherit">Logout</Button>
        </Toolbar>
      </AppBar>
    </div>
  );
}
