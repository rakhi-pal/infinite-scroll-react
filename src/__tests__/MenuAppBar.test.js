import { render } from '@testing-library/react';
import MenuAppBar from '../components/MenuAppBar/MenuAppBar';

test('renders learn react link', () => {
  const MenuAppBarComponent = render(<MenuAppBar />);
  expect(MenuAppBarComponent).toMatchSnapshot();
});