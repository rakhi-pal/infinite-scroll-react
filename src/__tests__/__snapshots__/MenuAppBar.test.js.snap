// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-menu-1"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <h6
              class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
            >
              Contacts
            </h6>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Logout
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-menu-1"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <h6
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
          >
            Contacts
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Logout
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
